extensions/ql-vscode/src/helpers.ts

```diff
@@ -29,8 +29,34 @@ export async function showAndLogErrorMessage(message: string, {
   items = [] as string[],
   fullMessage = undefined as (string | undefined)
 } = {}): Promise<string | undefined> {
-  return internalShowAndLog(message, items, outputLogger, Window.showErrorMessage, fullMessage);
+  return internalShowAndLog(shortenErrorMessage(message), items, outputLogger, Window.showErrorMessage, fullMessage);
 }
+
+/**
+ * Creates a succint version of an error message, e.g. for displaying in a pop-up.
+ * @param errorMessage The error message to shorten.
+ * @return A shortened version of the error message.
+ */
+function shortenErrorMessage(errorMessage: string): string {
+  // Filter out lines corresponding to Java stack traces.
+  const stackTraceLine = /\n\s*(com|org|net)\..*\.java:[0-9]+\)(?=\n)/g;
```

Comment:
Would this be better so you're not relying on newlines, which may not work on windows.

```suggestion
  const stackTraceLine = /^\s*(com|org|net)\..*\.java:[0-9]+\)$/g;
```